### PR TITLE
Add asset_genesis to asset TLV, split commitment root check, update proof file

### PR DESCRIPTION
**Pull request by guggero**
Fixes https://github.com/lightninglabs/taro/issues/145 and a bunch of smaller issues.

Also updates the proof file format according to https://github.com/lightninglabs/taro/pull/134.

### CODE REVIEW

**Comment by dstadulis** [comment]
@guggero is PR review the impediment for this being merged or are there other spec changes in the code that still need to be documented before review?

**Comment by guggero** [comment]
Just review. AFAIK there are no dependencies.

File: bip-taro-addr.mediawiki [hunk 1]
```diff
@@ -61,9 +61,16 @@ types:
 * type: 0 (<code>taro_version</code>)
 ** value:
 *** [<code>u8</code>:<code>version</code>]
-* type: 2 (<code>asset_id</code>)
+* type: 2 (<code>asset_genesis</code>)
 ** value:
-*** [<code>32*byte</code>:<code>asset_to_send</code>]
+*** [<code>32*byte</code>:<code>first_prev_out_hash</code>]
```
Comment (jharveyb) [inline bip-taro-addr.mediawiki:66]: 👍🏽

File: bip-taro-proof-file.mediawiki [hunk 2]
```diff
@@ -127,13 +130,16 @@ where:
...
+* <code>split_root_proof</code>: is an optional <code>taro_taproot_proof</code> that proves the inclusion of the split commitment's root asset in case of an asset split.
+* <code>taro_input_splits</code>: is an optional list of nested full proofs for any additional inputs found within the resulting asset.
 
 The final flat proof file has the following format:
-* <code>file_sum</code>: 32-bytes - 256 sum over the remaining contents of the file
-* <code>file_version</code>: 4-bytes - version of proof file
-* <code>taro_proofs</code>:<code>BigSize</code> length prefixed serialized proofs
-
-// TODO(roasbeef): make the file_sum instead an MMR over the set? allows for probabilistic verification of file, prove certain transition fragments
+* [<code>u32</code>:<code>file_version</code>] version of proof file
```
Comment (jharveyb) [inline bip-taro-proof-file.mediawiki:137]: Would be good to also update the proof verification pseudocode to reflect the per-proof checksum verification?

**Review by jharveyb — COMMENTED** [review]
Looks accurate, only note I have is the pseudocode update.

File: bip-taro-proof-file.mediawiki (hunk 2 above)
Comment (guggero) [reply bip-taro-proof-file.mediawiki:137]: Ah yes, good catch! I'm not very good with Python-like pseudo code, so I hope it's understandable what I wrote.

**Review by Roasbeef — APPROVED** [review]
LGTM 👒

**Pull request merged**